migrate/src/main/scala/migrate/Main.scala

```diff
@@ -75,17 +79,25 @@ object Main {
     scala3CompilerOptions: Seq[String]
   ): Try[Scala3Compiler] = {
     // It's easier no to deal with semanticdb option, since we don't need semanticdb files
-    val modified           = scala3CompilerOptions.filterNot(_ == "-Ysemanticdb")
+    // synthetics do not exist yet in scala 3 and not needed neither
+    val migrationEnabled = "source:3.0-migration"
```

Comment:
I would rather do that in the plugin because it might be surprising that `foo / migrate` works but `foo / compile` does not because `-source:3.0-migration` is missing from `foo / scalacOptions`:

```scala
val projectSettings = Seq(
  scalacOptions += {
   if (scalaVersion.value.startsWith("3.0")) Seq("-source:3.0-migration")
   else Seq.empty 
 }
) ++ ...
```

Reply:
ok